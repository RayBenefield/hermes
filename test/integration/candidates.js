import flame from '@leonardvandriel/flame';
import * as databases from './database'; // eslint-disable-line import/no-unresolved,import/extensions,no-unused-vars
import uuid from '../utils/mock-uuid';
import delay from '../utils/mock-delay';
import random from '../utils/mock-random';
import configureSave from '../utils/save-db';
import configureDomain from '../../src/domain';
import configureFlame from '../../src/lib/local-db';

const db = configureFlame(flame);
const domain = configureDomain({ db, uuid, random, delay });
const save = configureSave(flame); // eslint-disable-line no-unused-vars
beforeEach(() => flame.loadDatabase(undefined));

test.only('notify all players of vote starting', () => {
    // Given
    flame.loadDatabase(databases['candidates-shown']);

    // When
    return domain({
        action: 'candidates',
        payload: {
            game: '0-05526803409810505',
            round: '0-05526803409810505',
            notified: { 567890: true },
        },
    })
        .then(save('vote-started'))
        .then(results => expect({
            results,
            db: flame.get('/'),
        }).toMatchSnapshot());
});
