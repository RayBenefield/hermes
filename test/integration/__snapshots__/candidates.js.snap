// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notify all players of vote starting 1`] = `
Object {
  "db": Object {
    "candidates": Object {
      "0-05526803409810505": Object {
        "0-05526803409810505": Object {
          "candidates": Object {
            "123456": 268,
            "234567": 188,
            "345678": 175,
            "456789": 117,
            "567890": 31,
          },
          "game": "0-05526803409810505",
          "notified_players": Object {
            "123456": true,
            "234567": true,
            "345678": true,
            "456789": true,
            "567890": true,
          },
          "round": "0-05526803409810505",
        },
      },
    },
    "games": Object {
      "0-05526803409810505": Object {
        "id": "0-05526803409810505",
        "notified_players": Object {
          "123456": true,
          "234567": true,
          "345678": true,
          "456789": true,
          "567890": true,
        },
        "players": Object {
          "123456": true,
          "234567": true,
          "345678": true,
          "456789": true,
          "567890": true,
        },
        "rounds": Object {
          "0": "0-05526803409810505",
        },
      },
    },
    "hands": Object {
      "0-05526803409810505": Object {
        "123456": Object {
          "cards": Object {
            "1015": true,
            "1039": true,
            "1040": true,
            "310": true,
            "444": true,
            "597": true,
            "640": true,
            "714": true,
            "748": true,
          },
          "game": "0-05526803409810505",
          "player": 123456,
        },
        "234567": Object {
          "cards": Object {
            "218": true,
            "370": true,
            "429": true,
            "485": true,
            "624": true,
            "629": true,
            "725": true,
            "779": true,
            "951": true,
          },
          "game": "0-05526803409810505",
          "player": 234567,
        },
        "345678": Object {
          "cards": Object {
            "252": true,
            "322": true,
            "458": true,
            "582": true,
            "590": true,
            "609": true,
            "900": true,
            "965": true,
            "999": true,
          },
          "game": "0-05526803409810505",
          "player": 345678,
        },
        "456789": Object {
          "cards": Object {
            "241": true,
            "244": true,
            "334": true,
            "470": true,
            "498": true,
            "524": true,
            "593": true,
            "855": true,
            "898": true,
          },
          "game": "0-05526803409810505",
          "player": 456789,
        },
        "567890": Object {
          "cards": Object {
            "125": true,
            "227": true,
            "32": true,
            "36": true,
            "530": true,
            "597": true,
            "604": true,
            "605": true,
            "729": true,
          },
          "game": "0-05526803409810505",
          "player": 567890,
        },
      },
    },
    "players": Object {
      "facebook": Object {
        "123456": Object {
          "first_name": "Ray",
          "games": Object {
            "0-05526803409810505": true,
          },
          "id": 123456,
          "platform": "facebook",
        },
        "234567": Object {
          "first_name": "Hermon",
          "games": Object {
            "0-05526803409810505": true,
          },
          "id": 234567,
          "platform": "facebook",
        },
        "345678": Object {
          "first_name": "Hermes",
          "games": Object {
            "0-05526803409810505": true,
          },
          "id": 345678,
          "platform": "facebook",
        },
        "456789": Object {
          "first_name": "Hermshire",
          "games": Object {
            "0-05526803409810505": true,
          },
          "id": 456789,
          "platform": "facebook",
        },
        "567890": Object {
          "first_name": "Hermda",
          "games": Object {
            "0-05526803409810505": true,
          },
          "id": 567890,
          "platform": "facebook",
        },
      },
    },
    "queue": Object {},
    "rounds": Object {
      "0-05526803409810505": Object {
        "0-05526803409810505": Object {
          "candidates": Object {
            "123456": 268,
            "234567": 188,
            "345678": 175,
            "456789": 117,
            "567890": 31,
          },
          "card": Object {
            "contents": "I can only express myself through _.",
            "id": 189,
          },
          "game": "0-05526803409810505",
          "id": "0-05526803409810505",
        },
      },
    },
  },
  "results": Object {
    "action": "candidates",
    "candidates": Object {
      "123456": Object {
        "contents": "Dropping a chandelier on your enemies and riding the rope up.",
        "id": 268,
      },
      "234567": Object {
        "contents": "Keg stands.",
        "id": 188,
      },
      "345678": Object {
        "contents": "Skeletor.",
        "id": 175,
      },
      "456789": Object {
        "contents": "Sniffing glue.",
        "id": 117,
      },
      "567890": Object {
        "contents": "The forbidden fruit.",
        "id": 31,
      },
    },
    "game": Object {
      "id": "0-05526803409810505",
      "notified_players": Object {
        "123456": true,
        "234567": true,
        "345678": true,
        "456789": true,
        "567890": true,
      },
      "players": Object {
        "123456": true,
        "234567": true,
        "345678": true,
        "456789": true,
        "567890": true,
      },
      "rounds": Object {
        "0": "0-05526803409810505",
      },
    },
    "messages": Object {
      "123456": Array [
        Object {
          "payload": Object {
            "game": "0-05526803409810505",
            "round": "0-05526803409810505",
            "unranked": Array [
              Object {
                "contents": "Dropping a chandelier on your enemies and riding the rope up.",
                "id": 268,
              },
              Object {
                "contents": "Keg stands.",
                "id": 188,
              },
              Object {
                "contents": "Skeletor.",
                "id": 175,
              },
              Object {
                "contents": "Sniffing glue.",
                "id": 117,
              },
              Object {
                "contents": "The forbidden fruit.",
                "id": 31,
              },
            ],
          },
          "player": "123456",
          "type": "notify-candidates-ready-message",
        },
      ],
      "234567": Array [
        Object {
          "payload": Object {
            "game": "0-05526803409810505",
            "round": "0-05526803409810505",
            "unranked": Array [
              Object {
                "contents": "Dropping a chandelier on your enemies and riding the rope up.",
                "id": 268,
              },
              Object {
                "contents": "Keg stands.",
                "id": 188,
              },
              Object {
                "contents": "Skeletor.",
                "id": 175,
              },
              Object {
                "contents": "Sniffing glue.",
                "id": 117,
              },
              Object {
                "contents": "The forbidden fruit.",
                "id": 31,
              },
            ],
          },
          "player": "234567",
          "type": "notify-candidates-ready-message",
        },
      ],
      "345678": Array [
        Object {
          "payload": Object {
            "game": "0-05526803409810505",
            "round": "0-05526803409810505",
            "unranked": Array [
              Object {
                "contents": "Dropping a chandelier on your enemies and riding the rope up.",
                "id": 268,
              },
              Object {
                "contents": "Keg stands.",
                "id": 188,
              },
              Object {
                "contents": "Skeletor.",
                "id": 175,
              },
              Object {
                "contents": "Sniffing glue.",
                "id": 117,
              },
              Object {
                "contents": "The forbidden fruit.",
                "id": 31,
              },
            ],
          },
          "player": "345678",
          "type": "notify-candidates-ready-message",
        },
      ],
      "456789": Array [
        Object {
          "payload": Object {
            "game": "0-05526803409810505",
            "round": "0-05526803409810505",
            "unranked": Array [
              Object {
                "contents": "Dropping a chandelier on your enemies and riding the rope up.",
                "id": 268,
              },
              Object {
                "contents": "Keg stands.",
                "id": 188,
              },
              Object {
                "contents": "Skeletor.",
                "id": 175,
              },
              Object {
                "contents": "Sniffing glue.",
                "id": 117,
              },
              Object {
                "contents": "The forbidden fruit.",
                "id": 31,
              },
            ],
          },
          "player": "456789",
          "type": "notify-candidates-ready-message",
        },
      ],
    },
    "payload": Object {
      "game": "0-05526803409810505",
      "notified": Object {
        "567890": true,
      },
      "round": "0-05526803409810505",
    },
    "players": Array [
      Object {
        "first_name": "Ray",
        "games": Object {
          "0-05526803409810505": true,
        },
        "id": 123456,
        "platform": "facebook",
      },
      Object {
        "first_name": "Hermon",
        "games": Object {
          "0-05526803409810505": true,
        },
        "id": 234567,
        "platform": "facebook",
      },
      Object {
        "first_name": "Hermes",
        "games": Object {
          "0-05526803409810505": true,
        },
        "id": 345678,
        "platform": "facebook",
      },
      Object {
        "first_name": "Hermshire",
        "games": Object {
          "0-05526803409810505": true,
        },
        "id": 456789,
        "platform": "facebook",
      },
      Object {
        "first_name": "Hermda",
        "games": Object {
          "0-05526803409810505": true,
        },
        "id": 567890,
        "platform": "facebook",
      },
    ],
    "round": Object {
      "candidates": Object {
        "123456": 268,
        "234567": 188,
        "345678": 175,
        "456789": 117,
        "567890": 31,
      },
      "card": Object {
        "contents": "I can only express myself through _.",
        "id": 189,
      },
      "game": "0-05526803409810505",
      "id": "0-05526803409810505",
    },
    "unaddressedMessages": Array [
      Object {
        "payload": Object {
          "game": "0-05526803409810505",
          "round": "0-05526803409810505",
          "unranked": Array [
            Object {
              "contents": "Dropping a chandelier on your enemies and riding the rope up.",
              "id": 268,
            },
            Object {
              "contents": "Keg stands.",
              "id": 188,
            },
            Object {
              "contents": "Skeletor.",
              "id": 175,
            },
            Object {
              "contents": "Sniffing glue.",
              "id": 117,
            },
            Object {
              "contents": "The forbidden fruit.",
              "id": 31,
            },
          ],
        },
        "player": "123456",
        "type": "notify-candidates-ready-message",
      },
      Object {
        "payload": Object {
          "game": "0-05526803409810505",
          "round": "0-05526803409810505",
          "unranked": Array [
            Object {
              "contents": "Dropping a chandelier on your enemies and riding the rope up.",
              "id": 268,
            },
            Object {
              "contents": "Keg stands.",
              "id": 188,
            },
            Object {
              "contents": "Skeletor.",
              "id": 175,
            },
            Object {
              "contents": "Sniffing glue.",
              "id": 117,
            },
            Object {
              "contents": "The forbidden fruit.",
              "id": 31,
            },
          ],
        },
        "player": "234567",
        "type": "notify-candidates-ready-message",
      },
      Object {
        "payload": Object {
          "game": "0-05526803409810505",
          "round": "0-05526803409810505",
          "unranked": Array [
            Object {
              "contents": "Dropping a chandelier on your enemies and riding the rope up.",
              "id": 268,
            },
            Object {
              "contents": "Keg stands.",
              "id": 188,
            },
            Object {
              "contents": "Skeletor.",
              "id": 175,
            },
            Object {
              "contents": "Sniffing glue.",
              "id": 117,
            },
            Object {
              "contents": "The forbidden fruit.",
              "id": 31,
            },
          ],
        },
        "player": "345678",
        "type": "notify-candidates-ready-message",
      },
      Object {
        "payload": Object {
          "game": "0-05526803409810505",
          "round": "0-05526803409810505",
          "unranked": Array [
            Object {
              "contents": "Dropping a chandelier on your enemies and riding the rope up.",
              "id": 268,
            },
            Object {
              "contents": "Keg stands.",
              "id": 188,
            },
            Object {
              "contents": "Skeletor.",
              "id": 175,
            },
            Object {
              "contents": "Sniffing glue.",
              "id": 117,
            },
            Object {
              "contents": "The forbidden fruit.",
              "id": 31,
            },
          ],
        },
        "player": "456789",
        "type": "notify-candidates-ready-message",
      },
    ],
    "unnotifiedPlayers": Array [
      Object {
        "first_name": "Ray",
        "games": Object {
          "0-05526803409810505": true,
        },
        "id": 123456,
        "platform": "facebook",
      },
      Object {
        "first_name": "Hermon",
        "games": Object {
          "0-05526803409810505": true,
        },
        "id": 234567,
        "platform": "facebook",
      },
      Object {
        "first_name": "Hermes",
        "games": Object {
          "0-05526803409810505": true,
        },
        "id": 345678,
        "platform": "facebook",
      },
      Object {
        "first_name": "Hermshire",
        "games": Object {
          "0-05526803409810505": true,
        },
        "id": 456789,
        "platform": "facebook",
      },
    ],
  },
}
`;
